Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
I added it to make this code reusable from AutoFixture side. I'll need my own cache for already resolved results, so `CallResults` (more precisely, `CallResultsCache` wrapper) is ideal for that. However, AutoFixture sets results for the `ref` and `out` parameters, while method might be `void`. `CallResults` will just ignore such methods :(

I've decided to not change this behavior by default, because that could lead to performance degradation.